hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -171,4 +177,59 @@ private DatanodeStorageReport getStorageReport(DatanodeDetails datanode) {
     long committed = nodeStat.getCommitted().get();
     return new DatanodeStorageReport(capacity, used, remaining, committed);
   }
+
+  @PUT
+  @Path("/remove")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response removeDatanodes(List<String> uuids) {
+    List<DatanodeMetadata> notFoundDatanodes = new ArrayList<>();
+    List<DatanodeMetadata> removedDatanodes = new ArrayList<>();
+
+    Preconditions.checkNotNull(uuids, "Datanode list argument should not be null");
+    Preconditions.checkArgument(!uuids.isEmpty(), "Datanode list argument should not be empty");
+    try {
+      for (String uuid : uuids) {
+        DatanodeDetails nodeByUuid = nodeManager.getNodeByUuid(uuid);
+        try {
+          NodeStatus nodeStatus = nodeManager.getNodeStatus(nodeByUuid);
+          boolean isNodeDecommissioned = nodeByUuid.getPersistedOpState() == NodeOperationalState.DECOMMISSIONED;
+          boolean isNodeInMaintenance = nodeByUuid.getPersistedOpState() == NodeOperationalState.DECOMMISSIONED;
+          if (isNodeDecommissioned || isNodeInMaintenance || nodeStatus.isDead()) {
+            removedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+            nodeManager.removeNode(nodeByUuid);
+          } else {
+            Response.ResponseBuilder builder = Response.status(Response.Status.BAD_REQUEST);
+            builder.entity("Invalid request: Node: " + uuid + " should be in either DECOMMISSIONED or " +
+                "IN_MAINTENANCE mode or DEAD.");
+            // Build and return the response
+            return builder.build();
```

Comment:
Here, few of nodes are removed, but we are not returning those info. In this case, need return success nodes and failed nodes which is not matching for state
Or
First filter and return if any invalid nodes, and then return success nodes.

Reply:
Ok done.

Reply:
Still not fixed as return error from this point only. Need remove builder and return.